lib/mimic/cover.ex

```diff
@@ -22,21 +22,35 @@ defmodule Mimic.Cover do
   @doc false
   def replace_coverdata!(module, original_beam, original_coverdata_path) do
```

Comment:
It seems that most of this change here is related to the fact that reset is never called otherwise exported coverdata should be properly reimported :thinking: at the end of the test suite. 

We might want to set up an umbrella app as part of Mimic's test suite :thinking: 

Maybe a new umbrella app inside `umbrella_test` (or something similar) then add Mimic.copy on the same module in different apps, etc. Then finally implement the `test_coverage` like it's done here: https://github.com/edgurgel/mimic/blob/main/mix.exs#L19 (`test_coverage: UmbrellaTestCoverage`) 

then on CI it would `cd umbrella_test`, `mix test --cover` ? We could even simulate the case where one of the apps copy the module but never calls anything so the ExUnit callback to `Mimic.reset` is never called?

Where `UmbrellaTestCoverage` expects that the cover data is correct

Reply:
Yup I'm planning to try to get this done later in the week/weekend :+1: